lib/remote/apilistener-configsync.cpp

```diff
@@ -462,3 +477,53 @@ void ApiListener::SendRuntimeConfigObjects(const JsonRpcConnection::Ptr& aclient
 	Log(LogInformation, "ApiListener")
 		<< "Finished syncing runtime objects to endpoint '" << endpoint->GetName() << "'.";
 }
+
+/**
+ * Wait for the given object type and name to become available for processing
+ * and lock it again before releasing the lock.
+ *
+ * @param Type::Ptr ptype The type of the object you want to wait for
+ * @param String objName The object name you want to wait for
+ */
+void ObjectNameMutex::WaitForObject(const Type::Ptr& ptype, const String& objName)
```

Comment:
This mixes up the public interface and the implementation details in a way that's confusing, especially here:

> lock it again before releasing the lock

One time, "lock" seems to refer to the locking mechanism provided to the caller once and to the internal `m_Mutex` the other time.

Also, I'd just follow the typical Lock/Unlock naming for the functions, locking is the primary goal here and `std::mutex::lock()` also waits without explicitly having it in the name.